pkg/v3/plugin/hooks/add_from_staging.go

```diff
@@ -39,20 +53,44 @@ func (hook *AddFromStagingHook) RunHook(obs *ocr2keepersv3.AutomationObservation
 	if err != nil {
 		return err
 	}
-	// creating a map to hold the shuffled workIDs
-	shuffledIDs := make(map[string]string, len(results))
-	for _, result := range results {
-		shuffledIDs[result.WorkID] = random.ShuffleString(result.WorkID, rSrc)
+
+	results = hook.orderResults(results, rSrc)
+	n := len(results)
+	if n > limit {
```

Comment:
`n` isn't used outside the if so we can collapse line 58 and line 59 into `if n := len(results); n > limit`